ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetLogs.java

```diff
@@ -41,30 +43,26 @@ public String getName() {
   @Override
   public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     final FilterParameter filter = requestContext.getRequiredParameter(0, FilterParameter.class);
-    final LogsQuery query =
-        new LogsQuery.Builder().addresses(filter.getAddresses()).topics(filter.getTopics()).build();
 
-    if (isValid(filter)) {
+    if (!filter.isValid()) {
       return new JsonRpcErrorResponse(
           requestContext.getRequest().getId(), JsonRpcError.INVALID_PARAMS);
     }
-    if (filter.getBlockhash() != null) {
-      return new JsonRpcSuccessResponse(
-          requestContext.getRequest().getId(),
-          new LogsResult(blockchain.matchingLogs(filter.getBlockhash(), query)));
-    }
 
-    final long fromBlockNumber = filter.getFromBlock().getNumber().orElse(0);
-    final long toBlockNumber = filter.getToBlock().getNumber().orElse(blockchain.headBlockNumber());
+    final List<LogWithMetadata> matchingLogs =
+        filter
+            .getBlockHash()
+            .map(blockHash -> blockchain.matchingLogs(blockHash, filter.getLogsQuery()))
+            .orElseGet(
+                () -> {
+                  final long fromBlockNumber = filter.getFromBlock().getNumber().orElse(0);
```

Comment:
I think we should update this to be the headBlockNumber as well. This seems to be the "default" for Geth APIs. And probably update the `PrivGetLogs` as well.

I'm happy to do this in another PR is you don't wanna mix up with the other changes! :)

https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getlogs